File: Laba2/src/AndroidTV.java
/**
 * Class <code>AndroidTV</code> implements AndroidTV OS of television
 */
public class AndroidTV {

    enum Apps {
        HOME, NETFLIX, YOUTUBE, MEGOGO
    };

    private Apps app;
    private String wifiSSID;
    private String wifiPassword;
    private boolean isWifiConnected;

        /**
     * Constructor
     */
    public AndroidTV() {
        app = Apps.HOME;
        wifiSSID = "test";
        wifiPassword = "test";
        isWifiConnected = false;
    }

    /**
     * Constructor
     * 
     * @param ssid WIFI SSID
     * @param ssid WIFI password
     */
    public AndroidTV(String ssid, String password) {
        app = Apps.HOME;
        wifiSSID = ssid;
        wifiPassword = password;
        isWifiConnected = false;
    }
    
    /** 
     * Return current opened app
     * @return Apps
     */
    public Apps getOpenedApp(){
        return app;
    }

    /** 
     * Method opens app by app id
     * @param appIndex app id
     */
    public String openApp(int appIndex) {
        if (isWifiConnected) {
            if (appIndex < 0) {
                appIndex = 0;
            } else if (appIndex > 3) {
                appIndex = 3;
            }
            app = Apps.values()[appIndex];
            return app + " opened successfully";
        }
        return app + " not opened";
    }

    /** 
     * Method is used for changing WIFI SSID
     * @param ssid
     */
    public void setWifiSSID(String ssid) {
        isWifiConnected = false;
        wifiSSID = ssid;
    }

    
    /** 
     * Method is used for changing WIFI password
     * @param password
     */
    public void setWifiPassword(String password) {
        isWifiConnected = false;
        wifiPassword = password;
    }

    /** 
     * Method is used for connecting to WIFI
     */
    public void connectToWifi() {
        if (!wifiSSID.isEmpty() && !wifiPassword.isEmpty())
            isWifiConnected = true;
        
    }

    /** 
     * Method returns wifi status
     */
    public String getWifiStatus(){
        if(isWifiConnected)
            return "Wifi connected";
        return "Wifi not connected";
    }
}

File: Laba9/src/AndroidTV.py
from enum import Enum

class AndroidTV:

    class Apps(Enum):
        HOME = 0
        NETFLIX = 1
        YOUTUBE = 2
        MEGOGO = 3

    #Constructor
    def __init__(self, ssid = "test", password = "test"):
        self.__app = self.Apps.HOME
        self.__wifiSSID = ssid
        self.__wifiPassword = password
        self.__isWifiConnected = False

    #Return current opened app
    def getOpenedApp(self):
        return self.__app.name;

    #Method opens app by app id
    def openApp(self, appIndex):
        if self.__isWifiConnected:
            if appIndex < 0:
                appIndex = 0
            elif appIndex > 3:
                appIndex = 3
            self.__app = self.Apps(appIndex);
            return self.__app.name + " opened successfully";
        return self.__app.name + " not opened";
    
    #Method is used for changing WIFI SSID
    def setWifiSSID(self, ssid):
        self.__isWifiConnected = False
        self.__wifiSSID = ssid

    #Method is used for changing WIFI password
    def setWifiPassword(self, password):
        self.__isWifiConnected = False
        self.__wifiPassword = password

    #Method is used for connecting to WIFI
    def connectToWifi(self):
        if len(self.__wifiSSID) != 0 and len(self.__wifiPassword) != 0:
            self.__isWifiConnected = True

    #Method returns wifi status
    def getWifiStatus(self):
        if self.__isWifiConnected:
            return "Wifi connected"
        return "Wifi not connected"